src/driver/DriverFactory.ts

```diff
@@ -27,48 +8,59 @@ export class DriverFactory {
     /**
      * Creates a new driver depend on a given connection's driver type.
      */
-    create(connection: DataSource): Driver {
-        const { type } = connection.options
+    create(dataSource: DataSource): Driver {
+        const { type } = dataSource.options
+        return new (this.getDriver(type))(dataSource)
+    }
+
+    private getDriver(type: DataSource["options"]["type"]): DriverConstructor {
```

Comment:
```suggestion
        return new (this.drivers[type])(dataSource)
    }

    private const drivers = new Map([
      ['mysql', async () => import("./mysql/MysqlDriver").default],
      ...
    ])
```
something like that, just a different flavor

Reply:
I appreciate the feedback. but unless I get any response from any of the repo maintainers, I really don't want to spend any more time on this.